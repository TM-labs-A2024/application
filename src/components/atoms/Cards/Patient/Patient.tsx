import { WarningIcon } from '@chakra-ui/icons'
import { Link } from '@chakra-ui/next-js'
import { Text, Tag, TagLeftIcon, TagLabel, Tooltip } from '@chakra-ui/react'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Patient({
  href,
  title,
  description,
  status,
  hospitalizationPlace
}: {
  href: string
  title: string
  description: string
  status?: string
  hospitalizationPlace?: string
}) {
  return (
    <Link as={NextLink} href={href} className="w-full" data-testid="patient-card">
      <div className="flex flex-row items-center justify-between">
        <div className="flex flex-col items-start gap-1">
          <Text fontSize="md" as="b">
            {title}
          </Text>
          <Text>{description}</Text>
        </div>
        {status === 'hospitalizado' && (
          <Tooltip label={hospitalizationPlace}>
            <Tag size="md" key={status} variant="subtle" colorScheme="gray">
              <TagLeftIcon boxSize="12px" as={WarningIcon} />
              <TagLabel>{status}</TagLabel>
            </Tag>
          </Tooltip>
        )}
        <Chevron />
      </div>
    </Link>
  )
}
