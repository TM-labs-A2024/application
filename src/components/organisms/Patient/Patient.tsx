'use client'
import { Text, Heading, Stack, Link } from '@chakra-ui/react'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'
import NextLink from 'next/link'
import React from 'react'

export default function Patient({
  patient
}: {
  patient: {
    id: number
    name: string
    birthdate: string
    email: string
    phone: string
  }
}) {
  // --- Hooks -----------------------------------------------------------------
  // --- END: Hooks ------------------------------------------------------------

  // --- Local state -----------------------------------------------------------
  // --- END: Local state ------------------------------------------------------

  // --- Refs ------------------------------------------------------------------
  // --- END: Refs -------------------------------------------------------------

  // --- Redux -----------------------------------------------------------------
  // --- END: Redux ------------------------------------------------------------

  // --- Side effects ----------------------------------------------------------
  // --- END: Side effects -----------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div className="mx-auto block h-screen w-screen overflow-hidden px-8 pt-8 lg:w-1/2">
      <div className="mb-8 flex w-full flex-row justify-between">
        <Heading as="h2" size="md" noOfLines={1}>
          Perfil
        </Heading>
        <Link as={NextLink} href="/perfil/editar">
          Editar
        </Link>
      </div>
      <Stack spacing={1} mb={6}>
        <Heading as="h3" size="md" noOfLines={1}>
          {patient.name}
        </Heading>
        <Text>CI: {patient.id.toLocaleString('es-ES')}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Fecha de nacimiento</h4>
        <Text className="font-medium">
          {format(new Date(patient.birthdate), "dd 'de' MMMM, yyyy", {
            locale: es
          })}
        </Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Correo electrónico</h4>
        <Text className="font-medium">{patient.email}</Text>
      </Stack>
      <Stack mb={6}>
        <h4 className="text-sm text-gray-600">Teléfono</h4>
        <Text className="font-medium">{patient.phone}</Text>
      </Stack>
    </div>
  )
}
