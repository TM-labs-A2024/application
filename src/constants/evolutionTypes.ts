export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
