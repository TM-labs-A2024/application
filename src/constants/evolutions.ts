export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
