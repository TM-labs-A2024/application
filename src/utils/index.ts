import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
