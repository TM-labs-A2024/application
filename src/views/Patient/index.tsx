import Patient from '@src/components/organisms/Patient'
import { Patient as PatientType } from '@src/types'
import React from 'react'

export default function PatientView({ patient }: { patient: PatientType }) {
  return patient && <Patient patient={patient} />
}
