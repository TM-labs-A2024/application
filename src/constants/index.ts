import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
