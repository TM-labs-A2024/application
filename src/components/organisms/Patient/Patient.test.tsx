import { patient } from '@src/constants'
import { render, screen } from '@test/utils/index'
import { format } from 'date-fns'
import { es } from 'date-fns/locale/es'

import '@testing-library/jest-dom'
import Patient from './Patient'
import 'next/navigation'

jest.mock('next/navigation')

const hospitalizedPatient = {
  ...patient,
  status: 'Hospitalizado'
}

describe('Organisms > Patient test', () => {
  test('The component shows the name, govId, birthdate, email and phoneNumber of the patient', () => {
    render(<Patient patient={patient} />)

    const idMatcher = (content: string) => content.includes(patient.govId)
    const patientId = screen.getByText(idMatcher)

    const emailMatcher = (content: string) => content.includes(patient.email)
    const patientEmail = screen.getByText(emailMatcher)

    expect(screen.getByText(`${patient.firstname} ${patient.lastname}`)).toBeInTheDocument()
    expect(patientId).toBeInTheDocument()
    expect(
      screen.getByText(
        format(new Date(patient.birthdate), "dd 'de' MMMM, yyyy", {
          locale: es
        })
      )
    ).toBeInTheDocument()
    expect(patientEmail).toBeInTheDocument()
    expect(screen.getByText(patient.phoneNumber)).toBeInTheDocument()
  })

  test('If the patient is hospitalized, the component shows the status', () => {
    render(<Patient patient={hospitalizedPatient} />)

    expect(screen.getByText(hospitalizedPatient.status)).toBeInTheDocument()
  })

  test("If the patient is not hospitalized, the component doesn't show the status", () => {
    render(<Patient patient={patient} />)

    expect(screen.queryByText('Hospitalizado')).not.toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<Patient patient={patient} />)

    expect(screen.getByTestId('patient')).toMatchSnapshot()
  })
})
