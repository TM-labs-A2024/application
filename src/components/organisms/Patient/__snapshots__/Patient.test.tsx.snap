// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Patient test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 lg:px-96 pt-8"
  data-testid="patient"
>
  <div
    class="my-8 flex flex-row items-center justify-start gap-4"
  >
    <button
      aria-label="back"
      class="chakra-button css-84vx7q"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          fill="currentColor"
        />
      </svg>
    </button>
    <p
      class="chakra-text font-medium css-0"
    >
      Ficha médica
    </p>
  </div>
  <div
    class="flex flex-row items-center justify-between"
  >
    <div
      class="chakra-stack css-1uriqkk"
    >
      <h2
        class="chakra-heading css-ypohwm"
      >
        José Feliciano
         
        Gutierrez García
      </h2>
      <p
        class="chakra-text css-0"
      >
        CI: 
        E99885561
      </p>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    class="chakra-divider css-svjswr"
  />
  <div
    class="chakra-stack css-1420a3n"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Fecha de nacimiento
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      11 de noviembre, 1999
    </p>
  </div>
  <div
    class="chakra-stack css-1j35777"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Correo electrónico
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      user@example.com
    </p>
  </div>
  <div
    class="chakra-stack css-1j35777"
  >
    <h4
      class="text-sm text-gray-600"
    >
      Teléfono
    </h4>
    <p
      class="chakra-text font-medium css-0"
    >
      +584127659814
    </p>
  </div>
</div>
`
